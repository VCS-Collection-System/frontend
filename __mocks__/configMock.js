export const getConfig = (URL, configUrl, config) => {
  if(URL.indexOf(configUrl) > -1 ) {
    const country = URL.substring(configUrl.length);
    const countryConfig = config[country] || null;
    return Promise.resolve({data: countryConfig});
  } else {
    return Promise.reject({data: null});
  }
}