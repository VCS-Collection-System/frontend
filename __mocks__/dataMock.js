export const employeeInfo = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth":"2021-10-07",
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const attachmentObject = {
  "originalFileName":"blob",
  "size":100,
  "contentType":"image/png",
  "s3BucketName":"mockBucket",
  "s3UUID":"mockUUID"
}

export const vaxHistory = {
  "id":0,
  "employee": employeeInfo,
  "submissionDate":"2021-10-04T15:12:14.875",
  "review":{
     "reviewerEmployeeId":"z2",
     "reviewerNotes":"",
     "reviewDate":"2021-10-05T14:04:29.296",
     "reviewOutcome":"ACCEPTED",
     "rejectReason":""
  },
  "submittedBy":null,
  "autoApproved":false,
  "lastUpdatedTime":null,
  "lastUpdatedBy":null,
  "vaccineBrand":"MODERNA",
  "vaccineAdministrationDate":"2021-10-04",
  "fullVaccinatedFlag":false,
  "vaccineShotNumber":2,
  "documentType":"VaccineDocument",
  "attachment": {}
}

const taskEmployeeObject = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth": {
    "year": "2021",
    "monthValue": "12",
    "dayOfMonth": "16"
  },
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const mockTask = {
  "task-input-data": {
    "documentList": {
      "com.redhat.vcs.model.VaccineDocumentList": {
        "documents": [ {
          "com.redhat.vcs.model.VaccineCardDocument": {
            "employee": {
              "com.redhat.vcs.model.Employee": taskEmployeeObject
            },
            "vaccineBrand": {
              "com.redhat.vcs.model.VaccineBrand": "MODERNA"
            },
            "attachment": {
              "com.redhat.vcs.model.Attachment": attachmentObject
            },
            "vaccineAdministrationDate": {
              "year": "2021",
              "monthValue": "12",
              "dayOfMonth": "16"
            },
            "lotNumber": "mockLot"
          } }
        }
      }
    }
  }
}
