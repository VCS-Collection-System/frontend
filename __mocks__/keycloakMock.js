module.exports = {
    token: "bogustoken", 
    authenticated: false,
    loadUserProfile: jest.fn( ()=> {return Promise.resolve({
        attributes: {
            workforceid: ["mockId"]
        }
    })} ),
    login: jest.fn(),
    logout: jest.fn(),
    hasRealmRole: jest.fn(),
    profile: {
        firstName: "mockFirst",
        lastName: "mockLast",
        email: "mock@email.com",
        attributes: {
            workforceid: ["mockId"],
            agency: ["USA"]
        }
    }
};
