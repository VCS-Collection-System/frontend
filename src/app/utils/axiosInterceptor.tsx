import axios from "axios";

const axiosInterceptor = axios.create({})

axiosInterceptor.interceptors.request.use(
    config => {
        config.headers['Authorization'] = 'Bearer ' + sessionStorage.getItem("token");
        return config;
    },
    error => {
        Promise.reject(error)
    });
export default axiosInterceptor;