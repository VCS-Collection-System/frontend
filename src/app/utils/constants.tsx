declare global {  
  interface Window {
    REACT_APP_PAM_URL_BASE: string;
    REACT_APP_PAM_CONTAINER_ID: string;
    REACT_APP_KEYCLOAK_COUNTRY: string;
    REACT_APP_KEYCLOAK_EMAIL: string;
    REACT_APP_KEYCLOAK_FIRST_NAME: string;
    REACT_APP_KEYCLOAK_ID: string;
    REACT_APP_KEYCLOAK_LAST_NAME: string;
    REACT_APP_MAINTENANCE_BANNER: boolean;
    REACT_APP_MAINTENANCE_BANNER_TEXT: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_FEATURE_FLAGS: Array<Record<string, any>>;
    REACT_APP_MAX_DOSES: Record<string, number>;
    REACT_APP_CONSENT_LABEL: string;
    REACT_APP_ALTERNATE_CONSENT_LABEL: string;
    REACT_APP_BLOCKED_COUNTRIES: Array<string>;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_GLOBAL_CONFIG: Record<string, any>;
  }
}

// PAM urls
export const PAM_BASE_URL = window.REACT_APP_PAM_URL_BASE; 
export const SUBMIT_VAX_URL = PAM_BASE_URL + '/attestation/vax';
export const SUBMIT_TEST_URL = PAM_BASE_URL + '/attestation/covid-test-result';
export const PAM_TASK_URL = PAM_BASE_URL + '/tasks/';
export const RESERVED_TASK_URL = PAM_TASK_URL + 'owners?status=Reserved&user=';
export const INPROGRESS_TASK_URL = PAM_TASK_URL + 'owners?status=InProgress&user=';
export const ALL_TASK_URL = PAM_TASK_URL + 'owners?status=Ready&pageSize=200';
export const PRIORITY_URL = PAM_BASE_URL + '/query/priority-results';
export const PRIORITY_CONFIRM_URL = PAM_BASE_URL + '/query/priority-result/';
export const ATTACHMENT_URL = PAM_BASE_URL + '/attestation/attachment';
export const EMPLOYEE_URL = PAM_BASE_URL + '/query/employee/';
export const EMPLOYEE_NAME_URL = PAM_BASE_URL + '/query/employee-name/';
export const EMPLOYEE_VAX_HISTORY_URL = PAM_BASE_URL + '/query/vax-document/';
export const EMPLOYEE_TEST_HISTORY_URL = PAM_BASE_URL + '/query/test-result-document/';
export const VAX_TASK_DOCUMENT_URL = PAM_BASE_URL + '/query/task/document/';
export const DELETE_VAX_URL = PAM_BASE_URL + '/attestation/delete-vax-record/';
export const CONFIG_URL = PAM_BASE_URL + '/query/country-configs/';
export const SAVE_CONFIG_URL = PAM_BASE_URL + '/attestation/country-configs';

// List of countries allow to use this app
export const BLOCKED_COUNTRIES = window.REACT_APP_BLOCKED_COUNTRIES || []

export const KONAMI_CODE = [ "ArrowUp", "ArrowUp", "ArrowDown", "ArrowDown", "ArrowLeft", "ArrowRight", "ArrowLeft", "ArrowRight", "KeyB", "KeyA"];

// Maintenance Banner
export const MAINTENANCE_BANNER = window.REACT_APP_MAINTENANCE_BANNER || false
export const MAINTENANCE_BANNER_TEXT = window.REACT_APP_MAINTENANCE_BANNER_TEXT || ""

// Keycloak claims
export const KEYCLOAK_COUNTRY = window.REACT_APP_KEYCLOAK_COUNTRY
export const KEYCLOAK_EMAIL = window.REACT_APP_KEYCLOAK_EMAIL
export const KEYCLOAK_FIRST_NAME = window.REACT_APP_KEYCLOAK_FIRST_NAME
export const KEYCLOAK_ID = window.REACT_APP_KEYCLOAK_ID
export const KEYCLOAK_LAST_NAME = window.REACT_APP_KEYCLOAK_LAST_NAME
export const FEATURE_FLAGS = window.REACT_APP_FEATURE_FLAGS || []

// Maximum doses allowed by vaccine manufacturer
export const MAX_DOSES = window.REACT_APP_MAX_DOSES || {}

export const CONSENT_LABEL = window.REACT_APP_CONSENT_LABEL
export const ALTERNATE_CONSENT_LABEL = window.REACT_APP_ALTERNATE_CONSENT_LABEL

// Global country config
export const GLOBAL_CONFIG = window.REACT_APP_GLOBAL_CONFIG || {};

