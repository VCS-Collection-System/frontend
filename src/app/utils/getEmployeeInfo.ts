import {
  KEYCLOAK_FIRST_NAME, 
  KEYCLOAK_LAST_NAME, 
  KEYCLOAK_EMAIL,
  KEYCLOAK_COUNTRY,
  KEYCLOAK_ID
} from "@app/utils/constants"; 

export function getEmployeeInfo(keycloak) {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const jp = require('jsonpath');
  const profile = keycloak.profile;
  const employeeFirstName = jp.query(profile, KEYCLOAK_FIRST_NAME)[0];
  const employeeLastName = jp.query(profile, KEYCLOAK_LAST_NAME)[0];
  const employeeEmail = jp.query(profile, KEYCLOAK_EMAIL)[0];
  const employeeCountry = jp.query(profile, KEYCLOAK_COUNTRY)[0];
  const employeeID = jp.query(profile, KEYCLOAK_ID)[0];

  const isApprover = () => {
    if(employeeCountry && 
      (keycloak.hasRealmRole("vcs-superapprover") || keycloak.hasRealmRole("vcs-approver-" + employeeCountry.toLowerCase()))){
        return true;
    }

    return false;
  }

  const isSuperApprover = () => {
    return keycloak.hasRealmRole("vcs-superapprover");
  }

  return {
    firstName: employeeFirstName, 
    lastName: employeeLastName,
    email: employeeEmail,
    country: employeeCountry,
    id: employeeID,
    isApprover: isApprover(),
    isSuperApprover: isSuperApprover()
  }
}
