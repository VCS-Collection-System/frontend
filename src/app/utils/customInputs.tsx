import { useEffect, useState } from "react";
import { KONAMI_CODE } from './constants'

const useInputEvent = () => {
  const [key, setKey] = useState(null);
  useEffect(() => {
    const keyDownHandler = ({ code }) => setKey(code);
    const keyUpHandler = () => setKey(null);
    window.addEventListener('keydown', keyDownHandler);
    window.addEventListener('keyup', keyUpHandler);
    return () => {
      window.removeEventListener("keydown", keyDownHandler);
      window.removeEventListener("keyup", keyUpHandler)
    }
  }, []);
  return key;
}

const useSecretCode = (secretCode) => {
  const [count, setCount] = useState(0);
  const [success, setSuccess] = useState(false);
  const key = useInputEvent();
  
  useEffect(() => {
    if (key == null) return;
    if (key !== secretCode[count]) {
      setCount(0);
      return;
    }
    
    setCount((state) => state + 1);
    if (count + 1 === secretCode.length) {
      setSuccess(!success);
      setCount(0);
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [key]); 
  
  return success;
};

export const useDarkMode = () => {
  return useSecretCode(KONAMI_CODE);
};
