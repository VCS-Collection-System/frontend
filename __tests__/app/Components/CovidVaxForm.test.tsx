import React from 'react';
import reactRouterDom from 'react-router-dom'
import { render, fireEvent } from '@testing-library/react'
import { screen } from '@testing-library/dom'
import userEvent from '@testing-library/user-event'
import { shallow } from 'enzyme';
import CovidVaxForm from '../../../src/app/Components/CovidVaxForm';
import mockKeycloak from "../../../__mocks__/keycloakMock"
import { getConfig } from "../../../__mocks__/configMock"
import { CONFIG_URL } from '@app/utils/constants';

jest.mock("@react-keycloak/web", () => ({
  useKeycloak: jest.fn(() => ({ keycloak: mockKeycloak })),
}))

jest.mock("@app/utils/constants", () => ({
  FEATURE_FLAGS: [],
  KEYCLOAK_COUNTRY: "attributes.agency[0]",
  KEYCLOAK_EMAIL: "email",
  KEYCLOAK_FIRST_NAME: "firstName",
  KEYCLOAK_LAST_NAME: "lastName",
  KEYCLOAK_ID: "attributes.workforceid[0]",
  MAX_DOSES: {"Pfizer": 2},
  CONFIG_URL: "config/"
}))

const config = {"default": {
    "allowPdf": false,
    "dateOfBirth": true,
    "consentCheckboxAlternate": false,
    "consentCheckbox": false,
    "covidTest": false,
    "customInput": true,
    "hrDashboard": true,
    "hrSearch": true,
    "cdc": false,
    "divoc": false,
    "greenPass": false,
    "recovery": false,
    "fakeFeature": true
  }, 
  "aus": {"consentCheckbox": true},
  "deu": {"consentCheckboxAlternate": true}
}

jest.mock('@app/utils/axiosInterceptor', () => ({
  get: (URL: string) => getConfig(URL, CONFIG_URL, config)
}))

jest.mock("react-router-dom");
const mockPush = jest.fn();
reactRouterDom.useHistory = jest.fn().mockReturnValue({push: mockPush});

describe('<CovidVaxForm />', ()=> {
    it('should be defined', () => {
        expect(CovidVaxForm).toBeDefined()
    })
    it("should render without crashing", () => {
        const wrapper = shallow(<CovidVaxForm />)
        expect(wrapper)
    })
    it("should match existing snapshot", () => {
        const wrapper = shallow(<CovidVaxForm />)
        expect(wrapper).toMatchSnapshot()
    })
    it("birth date should be after min birth date", async () => {
        render(<CovidVaxForm />)
        await new Promise(process.nextTick);
        fireEvent.change(screen.getByLabelText("date_of_birth"), {target: {value: "12-31-1920"}});
        fireEvent.blur(screen.getByLabelText("date_of_birth"));
        expect(screen.getByLabelText("date_of_birth")).toHaveAttribute("aria-invalid", "true");
    })
    it("should be after min dose date", () => {
        render(<CovidVaxForm />)
        fireEvent.change(screen.getByLabelText("shot-1-date"), {target: {value: "05-01-2020"}});
        fireEvent.blur(screen.getByLabelText("shot-1-date"));
        expect(screen.getByLabelText("shot-1-date")).toHaveAttribute("aria-invalid", "true");
    })
    it("should add optional dose when clicking Add Dose", () => {
        render(<CovidVaxForm />)
        expect(screen.getByTestId("add-dose")).toBeTruthy();
        userEvent.click(screen.getByTestId("add-dose"));
        expect(screen.getByTestId("remove-dose")).toBeTruthy();
    })
    it("should validate 2nd dose is after 1st dose", () => {
        render(<CovidVaxForm />)
        userEvent.click(screen.getByTestId("add-dose"));
        fireEvent.change(screen.getByLabelText("shot-1-date"), {target: {value: "01-02-2022"}});
        fireEvent.change(screen.getByLabelText("shot-2-date"), {target: {value: "01-01-2022"}});
        fireEvent.blur(screen.getByLabelText("shot-2-date"));
        expect(screen.getByLabelText("shot-2-date")).toHaveAttribute("aria-invalid", "true");
    })
    it("should remove optional dose when clicking Remove Dose", () => {
        render(<CovidVaxForm />)
        userEvent.click(screen.getByTestId("add-dose"));
        userEvent.click(screen.getByTestId("remove-dose"));
        expect(screen.queryByTestId("remove-dose")).toBeFalsy();
    })
    it("should accept without consent in non-consent countries", () => {
        render(<CovidVaxForm />)
        expect(screen.queryByLabelText("consent-checkbox")).toBeFalsy();
    })
    it("should require consent in consent countries", async () => {
        mockKeycloak.profile.attributes.agency[0] = "AUS";
        render(<CovidVaxForm />);
        await new Promise(process.nextTick);
        expect(screen.getByLabelText("consent-checkbox")).toBeTruthy();
    })
    it("should require consent in alternate consent countries", async () => {
        mockKeycloak.profile.attributes.agency[0] = "DEU";
        render(<CovidVaxForm />);
        await new Promise(process.nextTick);
        expect(screen.getByLabelText("consent-checkbox")).toBeTruthy();
        mockKeycloak.profile.attributes.agency[0] = "USA";
    })
    it("should be able to click Back to return to menu", () => {
        render(<CovidVaxForm />);
        fireEvent.click(screen.getByText("Back"));
        expect(mockPush).toHaveBeenCalledWith('/attestmenu');
    })
})
