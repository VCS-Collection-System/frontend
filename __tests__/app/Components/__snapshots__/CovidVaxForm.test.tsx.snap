// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CovidVaxForm /> should match existing snapshot 1`] = `
<Form
  onSubmit={[Function]}
>
  <Feature
    name="consentCheckbox"
  >
    <Component
      control={
        Object {
          "_defaultValues": Object {},
          "_fields": Object {},
          "_formState": Object {
            "dirtyFields": Object {},
            "errors": Object {},
            "isDirty": false,
            "isSubmitSuccessful": false,
            "isSubmitted": false,
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "submitCount": 0,
            "touchedFields": Object {},
          },
          "_formValues": Object {},
          "_getFieldArrayValue": [Function],
          "_getIsDirty": [Function],
          "_getWatch": [Function],
          "_isInAction": false,
          "_isMounted": false,
          "_names": Object {
            "array": Set {},
            "mount": Set {},
            "unMount": Set {},
            "watch": Set {},
          },
          "_proxyFormState": Object {
            "dirtyFields": false,
            "errors": false,
            "isDirty": false,
            "isValid": false,
            "isValidating": false,
            "touchedFields": false,
          },
          "_removeFields": [Function],
          "_shouldUnregister": undefined,
          "_subjects": Object {
            "array": ge {
              "observers": Array [],
            },
            "control": ge {
              "observers": Array [],
            },
            "state": ge {
              "observers": Array [],
            },
            "watch": ge {
              "observers": Array [],
            },
          },
          "_updateFieldArray": [Function],
          "_updateProps": [Function],
          "_updateValid": [Function],
          "_updateValues": [Function],
          "register": [Function],
          "unregister": [Function],
        }
      }
      name="consent"
      render={[Function]}
      rules={
        Object {
          "required": true,
        }
      }
    />
  </Feature>
  <Feature
    name="consentCheckboxAlternate"
  >
    <Component
      control={
        Object {
          "_defaultValues": Object {},
          "_fields": Object {},
          "_formState": Object {
            "dirtyFields": Object {},
            "errors": Object {},
            "isDirty": false,
            "isSubmitSuccessful": false,
            "isSubmitted": false,
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "submitCount": 0,
            "touchedFields": Object {},
          },
          "_formValues": Object {},
          "_getFieldArrayValue": [Function],
          "_getIsDirty": [Function],
          "_getWatch": [Function],
          "_isInAction": false,
          "_isMounted": false,
          "_names": Object {
            "array": Set {},
            "mount": Set {},
            "unMount": Set {},
            "watch": Set {},
          },
          "_proxyFormState": Object {
            "dirtyFields": false,
            "errors": false,
            "isDirty": false,
            "isValid": false,
            "isValidating": false,
            "touchedFields": false,
          },
          "_removeFields": [Function],
          "_shouldUnregister": undefined,
          "_subjects": Object {
            "array": ge {
              "observers": Array [],
            },
            "control": ge {
              "observers": Array [],
            },
            "state": ge {
              "observers": Array [],
            },
            "watch": ge {
              "observers": Array [],
            },
          },
          "_updateFieldArray": [Function],
          "_updateProps": [Function],
          "_updateValid": [Function],
          "_updateValues": [Function],
          "register": [Function],
          "unregister": [Function],
        }
      }
      name="consent"
      render={[Function]}
      rules={
        Object {
          "required": true,
        }
      }
    />
  </Feature>
  <FormGroup
    fieldId="employee-full-name"
    helperText="Make sure this matches your full name."
    label="Full Name"
  >
    <TextInput
      aria-label="employee-full-name"
      isDisabled={true}
      type="text"
    />
  </FormGroup>
  <Component
    control={
      Object {
        "_defaultValues": Object {},
        "_fields": Object {},
        "_formState": Object {
          "dirtyFields": Object {},
          "errors": Object {},
          "isDirty": false,
          "isSubmitSuccessful": false,
          "isSubmitted": false,
          "isSubmitting": false,
          "isValid": false,
          "isValidating": false,
          "submitCount": 0,
          "touchedFields": Object {},
        },
        "_formValues": Object {},
        "_getFieldArrayValue": [Function],
        "_getIsDirty": [Function],
        "_getWatch": [Function],
        "_isInAction": false,
        "_isMounted": false,
        "_names": Object {
          "array": Set {},
          "mount": Set {},
          "unMount": Set {},
          "watch": Set {},
        },
        "_proxyFormState": Object {
          "dirtyFields": false,
          "errors": false,
          "isDirty": false,
          "isValid": false,
          "isValidating": false,
          "touchedFields": false,
        },
        "_removeFields": [Function],
        "_shouldUnregister": undefined,
        "_subjects": Object {
          "array": ge {
            "observers": Array [],
          },
          "control": ge {
            "observers": Array [],
          },
          "state": ge {
            "observers": Array [],
          },
          "watch": ge {
            "observers": Array [],
          },
        },
        "_updateFieldArray": [Function],
        "_updateProps": [Function],
        "_updateValid": [Function],
        "_updateValues": [Function],
        "register": [Function],
        "unregister": [Function],
      }
    }
    defaultValue=""
    name="alt_first_name"
    render={[Function]}
  />
  <Component
    control={
      Object {
        "_defaultValues": Object {},
        "_fields": Object {},
        "_formState": Object {
          "dirtyFields": Object {},
          "errors": Object {},
          "isDirty": false,
          "isSubmitSuccessful": false,
          "isSubmitted": false,
          "isSubmitting": false,
          "isValid": false,
          "isValidating": false,
          "submitCount": 0,
          "touchedFields": Object {},
        },
        "_formValues": Object {},
        "_getFieldArrayValue": [Function],
        "_getIsDirty": [Function],
        "_getWatch": [Function],
        "_isInAction": false,
        "_isMounted": false,
        "_names": Object {
          "array": Set {},
          "mount": Set {},
          "unMount": Set {},
          "watch": Set {},
        },
        "_proxyFormState": Object {
          "dirtyFields": false,
          "errors": false,
          "isDirty": false,
          "isValid": false,
          "isValidating": false,
          "touchedFields": false,
        },
        "_removeFields": [Function],
        "_shouldUnregister": undefined,
        "_subjects": Object {
          "array": ge {
            "observers": Array [],
          },
          "control": ge {
            "observers": Array [],
          },
          "state": ge {
            "observers": Array [],
          },
          "watch": ge {
            "observers": Array [],
          },
        },
        "_updateFieldArray": [Function],
        "_updateProps": [Function],
        "_updateValid": [Function],
        "_updateValues": [Function],
        "register": [Function],
        "unregister": [Function],
      }
    }
    defaultValue=""
    name="alt_last_name"
    render={[Function]}
  />
  <Feature
    name="dateOfBirth"
  >
    <Component
      control={
        Object {
          "_defaultValues": Object {},
          "_fields": Object {},
          "_formState": Object {
            "dirtyFields": Object {},
            "errors": Object {},
            "isDirty": false,
            "isSubmitSuccessful": false,
            "isSubmitted": false,
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "submitCount": 0,
            "touchedFields": Object {},
          },
          "_formValues": Object {},
          "_getFieldArrayValue": [Function],
          "_getIsDirty": [Function],
          "_getWatch": [Function],
          "_isInAction": false,
          "_isMounted": false,
          "_names": Object {
            "array": Set {},
            "mount": Set {},
            "unMount": Set {},
            "watch": Set {},
          },
          "_proxyFormState": Object {
            "dirtyFields": false,
            "errors": false,
            "isDirty": false,
            "isValid": false,
            "isValidating": false,
            "touchedFields": false,
          },
          "_removeFields": [Function],
          "_shouldUnregister": undefined,
          "_subjects": Object {
            "array": ge {
              "observers": Array [],
            },
            "control": ge {
              "observers": Array [],
            },
            "state": ge {
              "observers": Array [],
            },
            "watch": ge {
              "observers": Array [],
            },
          },
          "_updateFieldArray": [Function],
          "_updateProps": [Function],
          "_updateValid": [Function],
          "_updateValues": [Function],
          "register": [Function],
          "unregister": [Function],
        }
      }
      defaultValue=""
      name="date_of_birth"
      render={[Function]}
      rules={
        Object {
          "pattern": /\\^\\(0\\[1-9\\]\\|1\\[0-2\\]\\)\\\\-\\(0\\[1-9\\]\\|1\\\\d\\|2\\\\d\\|3\\[01\\]\\)\\\\-\\(19\\|20\\)\\\\d\\{2\\}\\$/,
          "required": true,
          "validate": [Function],
        }
      }
    />
  </Feature>
  <Grid
    hasGutter={true}
  >
    <CovidVaxDose
      booster={false}
      control={
        Object {
          "_defaultValues": Object {},
          "_fields": Object {},
          "_formState": Object {
            "dirtyFields": Object {},
            "errors": Object {},
            "isDirty": false,
            "isSubmitSuccessful": false,
            "isSubmitted": false,
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "submitCount": 0,
            "touchedFields": Object {},
          },
          "_formValues": Object {},
          "_getFieldArrayValue": [Function],
          "_getIsDirty": [Function],
          "_getWatch": [Function],
          "_isInAction": false,
          "_isMounted": false,
          "_names": Object {
            "array": Set {},
            "mount": Set {},
            "unMount": Set {},
            "watch": Set {},
          },
          "_proxyFormState": Object {
            "dirtyFields": false,
            "errors": false,
            "isDirty": false,
            "isValid": false,
            "isValidating": false,
            "touchedFields": false,
          },
          "_removeFields": [Function],
          "_shouldUnregister": undefined,
          "_subjects": Object {
            "array": ge {
              "observers": Array [],
            },
            "control": ge {
              "observers": Array [],
            },
            "state": ge {
              "observers": Array [],
            },
            "watch": ge {
              "observers": Array [],
            },
          },
          "_updateFieldArray": [Function],
          "_updateProps": [Function],
          "_updateValid": [Function],
          "_updateValues": [Function],
          "register": [Function],
          "unregister": [Function],
        }
      }
      doseNumber={1}
      getValues={[Function]}
      isRequired={true}
      key="0"
      removeDose={[Function]}
      vendorOptions={
        Array [
          Object {
            "label": "AstraZeneca",
            "value": "AstraZeneca",
          },
          Object {
            "label": "Comirnaty",
            "value": "Comirnaty",
          },
          Object {
            "label": "Covaxin",
            "value": "Covaxin",
          },
          Object {
            "label": "Covishield",
            "value": "Covishield",
          },
          Object {
            "label": "Covovax",
            "value": "Covovax",
          },
          Object {
            "label": "Janssen",
            "value": "Janssen",
          },
          Object {
            "label": "Johnson & Johnson",
            "value": "Johnson",
          },
          Object {
            "label": "Moderna",
            "value": "Moderna",
          },
          Object {
            "label": "Novavax",
            "value": "Novavax",
          },
          Object {
            "label": "Oxford",
            "value": "Oxford",
          },
          Object {
            "label": "Pfizer",
            "value": "Pfizer",
          },
          Object {
            "label": "Sinopharm",
            "value": "Sinopharm",
          },
          Object {
            "label": "Sinovac-CoronaVac",
            "value": "Sinovac_CoronaVac",
          },
          Object {
            "label": "Spikevax",
            "value": "Spikevax",
          },
          Object {
            "label": "Sputnik",
            "value": "Sputnik",
          },
          Object {
            "label": "Vaxzevria",
            "value": "Vaxzevria",
          },
        ]
      }
    />
  </Grid>
  <Text
    component="a"
    data-testid="add-dose"
    onClick={[Function]}
  >
    <PlusIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />
     Add Dose
  </Text>
  <Feature
    name="proof"
  >
    <Component
      control={
        Object {
          "_defaultValues": Object {},
          "_fields": Object {},
          "_formState": Object {
            "dirtyFields": Object {},
            "errors": Object {},
            "isDirty": false,
            "isSubmitSuccessful": false,
            "isSubmitted": false,
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "submitCount": 0,
            "touchedFields": Object {},
          },
          "_formValues": Object {},
          "_getFieldArrayValue": [Function],
          "_getIsDirty": [Function],
          "_getWatch": [Function],
          "_isInAction": false,
          "_isMounted": false,
          "_names": Object {
            "array": Set {},
            "mount": Set {},
            "unMount": Set {},
            "watch": Set {},
          },
          "_proxyFormState": Object {
            "dirtyFields": false,
            "errors": false,
            "isDirty": false,
            "isValid": false,
            "isValidating": false,
            "touchedFields": false,
          },
          "_removeFields": [Function],
          "_shouldUnregister": undefined,
          "_subjects": Object {
            "array": ge {
              "observers": Array [],
            },
            "control": ge {
              "observers": Array [],
            },
            "state": ge {
              "observers": Array [],
            },
            "watch": ge {
              "observers": Array [],
            },
          },
          "_updateFieldArray": [Function],
          "_updateProps": [Function],
          "_updateValid": [Function],
          "_updateValues": [Function],
          "register": [Function],
          "unregister": [Function],
        }
      }
      defaultValue=""
      name="proof_type"
      render={[Function]}
      rules={
        Object {
          "required": true,
        }
      }
    />
    <Component
      control={
        Object {
          "_defaultValues": Object {},
          "_fields": Object {},
          "_formState": Object {
            "dirtyFields": Object {},
            "errors": Object {},
            "isDirty": false,
            "isSubmitSuccessful": false,
            "isSubmitted": false,
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "submitCount": 0,
            "touchedFields": Object {},
          },
          "_formValues": Object {},
          "_getFieldArrayValue": [Function],
          "_getIsDirty": [Function],
          "_getWatch": [Function],
          "_isInAction": false,
          "_isMounted": false,
          "_names": Object {
            "array": Set {},
            "mount": Set {},
            "unMount": Set {},
            "watch": Set {},
          },
          "_proxyFormState": Object {
            "dirtyFields": false,
            "errors": false,
            "isDirty": false,
            "isValid": false,
            "isValidating": false,
            "touchedFields": false,
          },
          "_removeFields": [Function],
          "_shouldUnregister": undefined,
          "_subjects": Object {
            "array": ge {
              "observers": Array [],
            },
            "control": ge {
              "observers": Array [],
            },
            "state": ge {
              "observers": Array [],
            },
            "watch": ge {
              "observers": Array [],
            },
          },
          "_updateFieldArray": [Function],
          "_updateProps": [Function],
          "_updateValid": [Function],
          "_updateValues": [Function],
          "register": [Function],
          "unregister": [Function],
        }
      }
      name="vax_image"
      render={[Function]}
      rules={
        Object {
          "required": true,
        }
      }
    />
  </Feature>
  <FormGroup
    fieldId="accept buttons"
  >
    <Button
      icon={
        <CheckCircleIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      }
      type="submit"
      variant="primary"
    >
      Accept
    </Button>
    <Button
      onClick={[Function]}
      variant="plain"
    >
      Back
    </Button>
  </FormGroup>
</Form>
`;
